### DISCUSSION TITLE
Associate File Extensions with (Editor) SyntaxHighlighters

### DISCUSSION THREAD

**Discussion by ahrbe1** [godotengine/godot-proposals#11195] [Scripting]
I've been writing an `EditorSyntaxHighlighter` in GDScript. Two issues have occurred to me during this process that I'd like to discuss:

1. Godot doesn't seem to have a way to associate a file extension with a syntax highlighter. It appears the only way to use my new highlighter is to manually select `Edit -> Syntax Highlighter -> My Syntax Highlighter` after opening a file. It would also be useful to have registered file extensions appear in the `File -> Open... -> Recognized File Types` dropdown and then auto-apply my highlighter when a file with a registered extension is opened.

2. In lieu of that, there appears to be no way to hook the `EditorInterface` when a new text file is opened (The thought being to check the file extension and then add the highlighter to the active editor). There is the `ScriptEditor.editor_script_changed` signal, but this only works for scripts and not for other non-script file types. For non-script file types, the `Script` argument is passed as `null`.

I did find [EditorSyntaxHighlighter._get_supported_languages](https://docs.godotengine.org/en/stable/classes/class_editorsyntaxhighlighter.html#class-editorsyntaxhighlighter-private-method-get-supported-languages) but it's not clear to me what string to return here. So that might be an area for documentation improvement.

**Comment by Calinou** [comment]
Pull request opened:

- https://github.com/godotengine/godot/pull/99495

**Reply by jkulawik** [reply]
The new description should also clarify whether there's an attempt to auto-assign a highlighter or if it only filters out the highlighter when opening incompatible files.

**Reply by Calinou** [reply]
Done, can you check the updated PR at the same address?

PS: In the future, please leave comments related to the PR on the pull request's page directly, so that subscribed users can get notifications.
